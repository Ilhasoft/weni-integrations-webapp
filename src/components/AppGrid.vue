<template>
  <div>
    <section id="app-grid">
      <p class="app-grid__title">{{ $t(section) }}</p>

      <div class="app-grid__content">
        <unnnic-card
          ref="unnnic-marketplace-card"
          class="app-grid__content__item"
          v-for="app in apps"
          v-bind:key="app.id"
          type="marketplace"
          :title="app.name"
          :description="app.description"
          :icon="app.icon"
          :id="app.id"
          :comments="`${app.comments_count} ${$t('comments')}`"
          :rating="app.rating.average || 0"
          :iconSrc="app.icon"
          :typeAction="type"
          :buttonAction="/* istanbul ignore next */ () => openAddModal(app.id, app.name)"
          clickable
          @openModal="openAppModal(app.code)"
        />
      </div>
    </section>

    <unnnic-modal
      ref="unnnic-add-modal"
      :showModal="showAddModal"
      :text="$t('installed_app_title')"
      scheme="feedback-green"
      modal-icon="check-circle-1-1"
      @close="closeAddModal"
    >
      <span slot="message" v-html="$t('installed_app_description')"></span>
      <unnnic-button
        ref="unnnic-add-modal-close-button"
        slot="options"
        type="terciary"
        @click="closeAddModal"
        >{{ $t('close') }}</unnnic-button
      >
      <unnnic-button
        ref="unnnic-add-modal-navigate-button"
        slot="options"
        type="primary"
        @click="navigateToMyApps()"
      >
        {{ $t('access_my_apps') }}
      </unnnic-button>
    </unnnic-modal>
  </div>
</template>

<script>
  import { mapActions } from 'vuex';
  export default {
    name: 'AppGrid',
    props: {
      section: {
        type: String,
        default: null,
        validator(value) {
          return (
            [
              'communication_channels',
              'attendance_managers',
              'configured_apps',
              'installed_apps',
            ].indexOf(value) !== -1
          );
        },
      },
      type: {
        type: String,
        validator(value) {
          return ['add', 'config', 'edit'].indexOf(value) !== -1;
        },
      },
    },
    data() {
      return {
        showAddModal: false,
        apps: [],
      };
    },
    async mounted() {
      const category = this.getCategory(this.section);
      this.loadApps({ category });
    },
    methods: {
      ...mapActions(['getAllAppTypes']),
      async loadApps(filter) {
        if (this.type == 'add') {
          const { data } = await this.getAllAppTypes(filter);
          this.apps = data;
        } else {
          this.apps = [
            {
              code: 'wwc',
              name: 'Weni Web Chat MOCK',
              description:
                'Ullamco occaecat et id cillum. Amet exercitation nisi amet fugiat mollit minim est. Officia in enim amet ipsum Lorem velit sint pariatur sunt magna cupidatat. Magna non ea qui nisi ut.s',
              summary: 'Ullamco occaecat et id cillum.',
              category: 'channel',
              bg_color: '#00DED333',
              icon: 'https://weni-sp-push-dev.s3.sa-east-1.amazonaws.com/svg/logo-wwc.svg',
              assets: [
                {
                  type: 'image_banner',
                  url: 'https://weni-sp-push-dev.s3.sa-east-1.amazonaws.com/svg/Ilustra%C3%A7%C3%A3o%20banner%201.png',
                  description: '',
                },
              ],
              usersCount: 25,
              rating: {
                average: 4.9,
              },
              commentsCount: 3,
              config: {
                title: 'Chat Title',
                subtitle: 'Chat subtitle',
                placeholder: 'Chat placeholder',
                fullScreenButton: false,
                unreadIndicator: false,
                keepHistory: false,
                avatarImage:
                  'https://weni-sp-push-dev.s3.sa-east-1.amazonaws.com/svg/iconfinder_13-whatsapp_4202050+1.svg',
                customCSS: '',
                primaryColor: '#009E96',
                messageDelay: 1,
              },
            },
          ];
        }
      },
      getCategory(section) {
        const categories = {
          communication_channels: 'channel',
          attendance_managers: 'ticket',
          configured_apps: null,
          installed_apps: null,
        };
        return categories[section];
      },
      openAddModal() {
        this.showAddModal = true;
      },
      closeAddModal() {
        this.showAddModal = false;
      },
      navigateToMyApps() {
        this.$router.push('apps');
      },
      openAppModal(code) {
        this.$router.push(`/apps/${code}/details`);
      },
    },
  };
</script>

<style lang="scss" scoped>
  .app-grid {
    &__title {
      color: $unnnic-color-neutral-darkest;
      font-size: $unnnic-font-size-title-sm;
    }

    &__content {
      display: grid;
      grid-template-columns: repeat(auto-fit, minmax(256px, 256px));
      grid-gap: $unnnic-spacing-stack-sm;
      align-items: flex-start;
    }
  }
</style>
