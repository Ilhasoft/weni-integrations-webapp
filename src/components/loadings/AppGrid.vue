<template>
  <div>
    <div class="title">
      <unnnic-skeleton-loading tag="div" width="40px" height="40px" />
      <unnnic-skeleton-loading tag="div" width="208px" height="40px" />
    </div>
    <div class="grid">
      <unnnic-skeleton-loading
        v-for="index in 4"
        :key="index"
        class="grid__item"
        tag="div"
        width="256px"
        height="200px"
      />
    </div>
  </div>
</template>

<script>
  // import SkeletonLoading from '../Skeleton/SkeletonLoading.vue';
  export default {};
</script>

<style lang="scss" scoped>
  .title {
    display: flex;
    gap: $unnnic-spacing-inline-xs;
    margin: $unnnic-spacing-stack-md 0;
  }
  .grid {
    display: grid;
    grid-template-columns: repeat(auto-fit, minmax(256px, 256px));
    grid-gap: $unnnic-spacing-stack-sm;
    align-items: flex-start;
  }
</style>
