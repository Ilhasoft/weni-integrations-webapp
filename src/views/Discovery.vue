<template>
  <div>
    <app-grid
      section="channel"
      type="add"
      :loading="loadingAllAppTypes"
      :apps="allAppTypes"
      @update="fetchChannels"
    />
  </div>
</template>
<script>
  import AppGrid from '../components/AppGrid.vue';
  import { mapActions, mapGetters, mapState } from 'vuex';
  import { unnnicCallAlert } from '@weni/unnnic-system';

  export default {
    name: 'Discovery',
    components: { AppGrid },
    data() {
      return {
        channels: {
          loading: true,
          data: null,
        },
      };
    },
    async mounted() {
      await this.fetchChannels();
    },
    computed: {
      ...mapGetters({
        getSelectedProject: 'getSelectedProject',
      }),
      ...mapState({
        allAppTypes: (state) => state.appType.allAppTypes,
        loadingAllAppTypes: (state) => state.appType.loadingAllAppTypes,
        errorAllAppTypes: (state) => state.appType.errorAllAppTypes,
      }),
    },
    methods: {
      ...mapActions(['getAllAppTypes']),
      async fetchChannels() {
        const params = {
          category: 'channel',
        };
        await this.getAllAppTypes({ params });

        if (this.errorAllAppTypes) {
          unnnicCallAlert({
            props: {
              text: this.$t('apps.discovery.fetch_error'),
              title: 'Error',
              icon: 'alert-circle-1',
              scheme: 'feedback-red',
              position: 'bottom-right',
              closeText: this.$t('general.Close'),
            },
            seconds: 6,
          });
          return;
        }

        this.channels.data = this.allAppTypes.reverse();
      },
    },
  };
</script>
<style lang="scss"></style>
