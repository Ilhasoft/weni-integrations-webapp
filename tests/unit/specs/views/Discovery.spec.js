import { unnnicCallAlert as mockUnnnicCallAlert } from '@weni/unnnic-system';

jest.mock('@weni/unnnic-system', () => ({
  ...jest.requireActual('@weni/unnnic-system'),
  unnnicCallAlert: jest.fn(),
}));

import Vuex from 'vuex';
import { mount, createLocalVue } from '@vue/test-utils';
import Discovery from '@/views/Discovery/index.vue';
import AppGrid from '@/components/AppGrid/index.vue';
import { singleApp } from '../../../__mocks__/appMock';

const genericApp = {
  ...singleApp,
  name: 'generic',
  code: 'generic',
  generic: true,
  config: { channel_name: 'A random generic' },
};

const mockManuallyCreateApp = jest.fn();
AppGrid.methods = {
  ...AppGrid.methods,
  manuallyCreateApp: mockManuallyCreateApp,
};

const mountComponent = async ({ createAppCode = null, apps = [singleApp] } = {}) => {
  const actions = {
    getAllAppTypes: jest.fn(),
    deleteApp: jest.fn(),
    getConfiguredApps: jest.fn(),
  };

  const state = {
    appType: {
      loadingDeleteApp: false,
      errorDeleteApp: false,

      allAppTypes: apps,
      loadingAllAppTypes: false,
      errorAllAppTypes: false,
    },
    auth: {
      project: '123',
    },
  };

  const externalServicesActions = {
    getExternalServicesTypes: jest.fn(),
  };

  const externalServicesState = {
    loadingExternalServices: false,
    errorExternalServices: null,
    externalServicesList: apps,
  };

  const store = new Vuex.Store({
    modules: {
      externals: {
        namespaced: true,
        actions: externalServicesActions,
        state: externalServicesState,
      },
    },
    actions,
    state,
  });

  const wrapper = mount(Discovery, {
    localVue,
    store,
    mocks: {
      $t: () => 'some specific text',
      $route: {
        query: { create_app: createAppCode },
      },
    },
    stubs: {
      AppGrid,
    },
  });

  await wrapper.vm.$nextTick();
  await jest.runAllTimers();

  return { wrapper, store, actions };
};

const localVue = createLocalVue();
localVue.use(Vuex);

describe('Discovery/index.vue', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  it('should be rendered properly', async () => {
    const { wrapper } = await mountComponent();
    expect(wrapper).toMatchSnapshot();
  });

  describe('mounted', () => {
    it('should call fetchChannels', async () => {
      const spy = jest.spyOn(Discovery.methods, 'fetchChannels');
      await mountComponent();
      expect(spy).toHaveBeenCalledTimes(1);
    });

    it('should call manuallyCreateApp if create_app is recieved as query param', async () => {
      await mountComponent({ createAppCode: 'tg' });

      expect(mockManuallyCreateApp).toHaveBeenCalledTimes(1);
    });

    it('should not call manuallyCreateApp if create_app is not recieved as query param', async () => {
      await mountComponent({ createAppCode: null });

      expect(mockManuallyCreateApp).toHaveBeenCalledTimes(0);
    });
  });

  describe('fetchChannels', () => {
    beforeEach(() => {
      jest.resetAllMocks();
    });

    it('should call getAllAppTypes', async () => {
      const { wrapper, actions } = await mountComponent();
      jest.clearAllMocks();
      expect(actions.getAllAppTypes).not.toHaveBeenCalled();
      await wrapper.vm.fetchChannels();
      expect(actions.getAllAppTypes).toHaveBeenCalledTimes(1);
    });

    it('should call unnnicCallAlert on error', async () => {
      const { wrapper, store } = await mountComponent();
      store.state.appType.errorAllAppTypes = true;
      expect(mockUnnnicCallAlert).not.toHaveBeenCalled();
      await wrapper.vm.fetchChannels();
      expect(mockUnnnicCallAlert).toHaveBeenCalledTimes(1);
    });
  });

  describe('search', () => {
    it('should filter apps, externalservices and biApps apps by name', async () => {
      const { wrapper, store } = await mountComponent();
      store.state.appType.allAppTypes = [singleApp, genericApp];
      store.state.externals.externalServicesList = [singleApp, genericApp];

      expect(wrapper.vm.filteredApps).toEqual([singleApp, genericApp]);
      expect(wrapper.vm.filteredExternalServices).toEqual([singleApp, genericApp]);
      expect(wrapper.vm.filteredBiApps).toEqual(wrapper.vm.biApps);

      wrapper.vm.searchTerm = 'weni';
      expect(wrapper.vm.filteredApps).toEqual([singleApp]);
      expect(wrapper.vm.filteredExternalServices).toEqual([singleApp]);
      expect(wrapper.vm.filteredBiApps).toEqual([]);

      wrapper.vm.searchTerm = ' ';
      expect(wrapper.vm.filteredApps).toEqual([singleApp, genericApp]);
      expect(wrapper.vm.filteredExternalServices).toEqual([singleApp, genericApp]);
      expect(wrapper.vm.filteredBiApps).toEqual(wrapper.vm.biApps);
    });

    it('should return empty array if there were no apps from store', async () => {
      const { wrapper, store } = await mountComponent();
      store.state.appType.allAppTypes = null;
      store.state.externals.externalServicesList = null;

      expect(wrapper.vm.filteredApps).toEqual([]);
      expect(wrapper.vm.filteredExternalServices).toEqual([]);
    });
  });
});
