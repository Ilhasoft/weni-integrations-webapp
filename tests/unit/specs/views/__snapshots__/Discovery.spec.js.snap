// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Discovery/index.vue should be rendered properly 1`] = `
<div class="discovery-content">
  <div class="unnnic-autocomplete discovery-content__search">
    <!---->
    <div data-v-400546b3="" class="text-input size--md has-icon-left" v-bind="[object Object]" tooltip-side-icon-right="right"><input data-v-97b44e8c="" data-v-400546b3="" type="text" size="0" v-bind="[object Object]" tooltip-side-icon-right="right" placeholder="some specific text" iconleft="search-1" class="input-itself input size-md normal"><span data-v-4ce83152="" data-v-400546b3="" class="unnnic-icon unnnic-icon__size--sm unnnic-icon-scheme--neutral-cloudy icon-left"><svg viewBox="0 0 40 40" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M34.0615 35.0005C33.8116 35.0005 33.5754 34.903 33.3992 34.7256L25.2575 26.5826C24.3065 27.3961 23.2355 28.0597 22.067 28.5559C20.5312 29.2082 18.9178 29.5394 17.272 29.5394C15.7011 29.5394 14.1578 29.2382 12.6819 28.6421C9.64389 27.4161 7.26574 25.0817 5.98481 22.0662C4.70388 19.052 4.67263 15.7178 5.89858 12.6811C7.12452 9.64433 9.45894 7.26492 12.4732 5.98399C14.009 5.33041 15.6211 5.00049 17.2682 5.00049C18.8391 5.00049 20.3825 5.30291 21.8583 5.89776C24.8963 7.12371 27.2745 9.45812 28.5554 12.4736C29.8363 15.4891 29.8676 18.822 28.6416 21.8588C28.1393 23.1035 27.4469 24.2444 26.5809 25.2554L34.7251 33.4009C35.09 33.7658 35.09 34.3606 34.7251 34.7268C34.5477 34.903 34.3115 35.0005 34.0615 35.0005ZM17.2682 6.87502C15.8748 6.87502 14.5077 7.15495 13.2068 7.70856C10.6524 8.79454 8.67413 10.809 7.63564 13.3821C6.59715 15.9552 6.6234 18.7783 7.70813 21.3327C8.7941 23.887 10.8086 25.8653 13.3817 26.9038C14.6326 27.4086 15.9411 27.6636 17.2707 27.6636C18.6641 27.6636 20.0313 27.3836 21.3322 26.83C22.5494 26.3127 23.6404 25.5828 24.5739 24.6606C24.5864 24.6431 24.6002 24.6281 24.6139 24.6143C24.6339 24.5943 24.6526 24.5781 24.6701 24.5656C25.6236 23.6021 26.381 22.4499 26.9021 21.1565C27.9406 18.5833 27.9143 15.7603 26.8283 13.2059C25.7436 10.6516 23.7279 8.67332 21.1548 7.63483C19.9063 7.13121 18.5979 6.87502 17.2682 6.87502Z" fill="#3B414D" class="primary"></path></svg></span>
      <!---->
    </div>
    <div class="unnnic-autocomplete__container-list" style="display: none;">
      <div class="unnnic-autocomplete__list unnnic-autocomplete__list--size-md">
        <div class="unnnic--clickable option">
          <div title="Weni Web Chat" class="label">Weni Web Chat</div>
        </div>
        <div class="unnnic--clickable option">
          <div title="Weni Web Chat" class="label">Weni Web Chat</div>
        </div>
        <div class="unnnic--clickable option">
          <div title="Power BI" class="label">Power BI</div>
        </div>
      </div>
    </div>
    <!---->
  </div>
  <!---->
  <div class="discovery-content__grids">
    <div>
      <section id="app-grid">
        <div class="app-grid__header">
          <div data-v-0231bb98="" class="app-grid__header__icon unnnic-avatar-icon aux-purple-500 sm opacity"><span data-v-4ce83152="" data-v-0231bb98="" class="material-symbols-rounded unnnic-icon-scheme--aux-purple-500 unnnic-icon-size--md material-symbols-rounded--filled"> sms </span></div>
          <p class="app-grid__header__title">some specific text</p>
        </div>
        <div class="app-grid__content">
          <div class="unnnic-card-component unnnic-card-marketplace unnnic-card-marketplace__clickable unnnic-card-marketplace__add app-grid__content__item" v-bind="[object Object]" button-action="() => {}" percent="0" info-position="right" enabled="enabled" scheme="aux-blue" has-information-icon="true">
            <div class="unnnic-card-marketplace__icons"><img src="https://weni-sp-push-dev.s3.sa-east-1.amazonaws.com/svg/logo-wwc.svg" class="unnnic-card-marketplace__icons__app">
              <div><button data-v-2d9f2cbf="" disabled="disabled" class="unnnic-button unnnic-button--size-small unnnic-button--secondary unnnic-button--icon-on-center loading-button">
                  <!---->
                  <!----><span data-v-4ce83152="" data-v-2d9f2cbf="" class="unnnic-icon unnnic-icon__size--sm unnnic-icon-scheme--neutral-clean"><svg viewBox="0 0 40 40" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M20 35.0005C19.4825 35.0005 19.0625 34.5805 19.0625 34.063V20.938H5.9375C5.42 20.938 5 20.518 5 20.0005C5 19.483 5.42 19.063 5.9375 19.063H19.0625V5.93799C19.0625 5.42049 19.4825 5.00049 20 5.00049C20.5175 5.00049 20.9375 5.42049 20.9375 5.93799V19.063H34.0625C34.58 19.063 35 19.483 35 20.0005C35 20.518 34.58 20.938 34.0625 20.938H20.9375V34.063C20.9375 34.5805 20.5175 35.0005 20 35.0005Z" fill="#3B414D" class="primary"></path></svg></span><span data-v-2d9f2cbf="" class="unnnic-button__label">  </span>
                  <!----></button>
                <!---->
                <!---->
              </div>
            </div>
            <div class="unnnic-card-marketplace__content">
              <div class="unnnic-card-marketplace__content__title">Weni Web Chat</div>
              <div title="some specific text" class="unnnic-card-marketplace__content__description"> some specific text </div>
              <div class="unnnic-card-marketplace__content__rating"><span data-v-4ce83152="" class="unnnic-card-marketplace__content__rating__star unnnic-icon unnnic-icon__size--ant unnnic-icon-scheme--feedback-yellow"><svg viewBox="0 0 40 40" xmlns="http://www.w3.org/2000/svg"><path d="M34.752 15.8274C34.6157 15.461 34.3708 15.145 34.0501 14.9215C33.7294 14.698 33.3482 14.5776 32.9573 14.5764H25.1872C25.0549 14.5765 24.926 14.5355 24.818 14.4592C24.7101 14.3828 24.6285 14.2748 24.5847 14.1501L21.7955 6.24466C21.6589 5.87927 21.4139 5.56433 21.0934 5.34198C20.7729 5.11964 20.3921 5.00049 20.002 5.00049C19.6119 5.00049 19.2311 5.11964 18.9106 5.34198C18.5901 5.56433 18.3451 5.87927 18.2085 6.24466L18.2021 6.26508L15.4193 14.1501C15.3755 14.2746 15.2942 14.3824 15.1865 14.4588C15.0788 14.5352 14.9501 14.5763 14.8181 14.5764H7.04544C6.65205 14.5761 6.2681 14.697 5.94584 14.9226C5.62357 15.1483 5.37862 15.4677 5.24431 15.8374C5.10999 16.2072 5.09282 16.6093 5.19514 16.9892C5.29746 17.3691 5.5143 17.7082 5.81617 17.9604L12.4336 23.4494C12.5326 23.5316 12.6044 23.6417 12.6397 23.7654C12.675 23.8891 12.6721 24.0205 12.6314 24.1426L9.85121 32.4794C9.72272 32.8649 9.72018 33.2812 9.84396 33.6682C9.96774 34.0552 10.2114 34.3928 10.5397 34.6322C10.8681 34.8715 11.264 35.0003 11.6704 34.9997C12.0767 34.9991 12.4723 34.8693 12.7999 34.6291L19.6241 29.6252C19.7336 29.545 19.8657 29.5018 20.0014 29.5018C20.137 29.5018 20.2691 29.545 20.3786 29.6252L27.2002 34.6278C27.5277 34.8689 27.9235 34.9994 28.3302 35.0005C28.7369 35.0016 29.1334 34.8731 29.4621 34.6338C29.7909 34.3944 30.035 34.0566 30.1589 33.6693C30.2829 33.2819 30.2803 32.8652 30.1515 32.4794L27.3713 24.1375C27.3306 24.0154 27.3277 23.884 27.363 23.7603C27.3983 23.6366 27.4701 23.5265 27.5691 23.4443L34.1993 17.9477C34.4973 17.6933 34.7101 17.3536 34.8089 16.9744C34.9078 16.5953 34.8879 16.1949 34.752 15.8274V15.8274Z" class="primary"></path></svg></span>
                <div class="unnnic-card-marketplace__content__rating__value">4.9</div>
                <div class="unnnic-card-marketplace__content__rating__comments">(undefined some specific text)</div>
              </div>
            </div>
          </div>
        </div>
        <!---->
      </section>
      <!---->
      <div>
        <transition-stub name="fade">
          <!---->
        </transition-stub>
        <!---->
      </div>
    </div>
    <div>
      <section id="app-grid">
        <div class="app-grid__header">
          <div data-v-0231bb98="" class="app-grid__header__icon unnnic-avatar-icon aux-blue-500 sm opacity"><span data-v-4ce83152="" data-v-0231bb98="" class="material-symbols-rounded unnnic-icon-scheme--aux-blue-500 unnnic-icon-size--md material-symbols-rounded--filled"> local_mall </span></div>
          <p class="app-grid__header__title">some specific text</p>
        </div>
        <div class="app-grid__content">
          <div class="unnnic-card-component unnnic-card-marketplace unnnic-card-marketplace__clickable unnnic-card-marketplace__add app-grid__content__item" v-bind="[object Object]" button-action="() => {}" percent="0" info-position="right" enabled="enabled" scheme="aux-blue" has-information-icon="true">
            <div class="unnnic-card-marketplace__icons"><img src="https://weni-sp-push-dev.s3.sa-east-1.amazonaws.com/svg/logo-wwc.svg" class="unnnic-card-marketplace__icons__app">
              <div><button data-v-2d9f2cbf="" disabled="disabled" class="unnnic-button unnnic-button--size-small unnnic-button--secondary unnnic-button--icon-on-center loading-button">
                  <!---->
                  <!----><span data-v-4ce83152="" data-v-2d9f2cbf="" class="unnnic-icon unnnic-icon__size--sm unnnic-icon-scheme--neutral-clean"><svg viewBox="0 0 40 40" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M20 35.0005C19.4825 35.0005 19.0625 34.5805 19.0625 34.063V20.938H5.9375C5.42 20.938 5 20.518 5 20.0005C5 19.483 5.42 19.063 5.9375 19.063H19.0625V5.93799C19.0625 5.42049 19.4825 5.00049 20 5.00049C20.5175 5.00049 20.9375 5.42049 20.9375 5.93799V19.063H34.0625C34.58 19.063 35 19.483 35 20.0005C35 20.518 34.58 20.938 34.0625 20.938H20.9375V34.063C20.9375 34.5805 20.5175 35.0005 20 35.0005Z" fill="#3B414D" class="primary"></path></svg></span><span data-v-2d9f2cbf="" class="unnnic-button__label">  </span>
                  <!----></button>
                <!---->
                <!---->
              </div>
            </div>
            <div class="unnnic-card-marketplace__content">
              <div class="unnnic-card-marketplace__content__title">Weni Web Chat</div>
              <div title="some specific text" class="unnnic-card-marketplace__content__description"> some specific text </div>
              <div class="unnnic-card-marketplace__content__rating"><span data-v-4ce83152="" class="unnnic-card-marketplace__content__rating__star unnnic-icon unnnic-icon__size--ant unnnic-icon-scheme--feedback-yellow"><svg viewBox="0 0 40 40" xmlns="http://www.w3.org/2000/svg"><path d="M34.752 15.8274C34.6157 15.461 34.3708 15.145 34.0501 14.9215C33.7294 14.698 33.3482 14.5776 32.9573 14.5764H25.1872C25.0549 14.5765 24.926 14.5355 24.818 14.4592C24.7101 14.3828 24.6285 14.2748 24.5847 14.1501L21.7955 6.24466C21.6589 5.87927 21.4139 5.56433 21.0934 5.34198C20.7729 5.11964 20.3921 5.00049 20.002 5.00049C19.6119 5.00049 19.2311 5.11964 18.9106 5.34198C18.5901 5.56433 18.3451 5.87927 18.2085 6.24466L18.2021 6.26508L15.4193 14.1501C15.3755 14.2746 15.2942 14.3824 15.1865 14.4588C15.0788 14.5352 14.9501 14.5763 14.8181 14.5764H7.04544C6.65205 14.5761 6.2681 14.697 5.94584 14.9226C5.62357 15.1483 5.37862 15.4677 5.24431 15.8374C5.10999 16.2072 5.09282 16.6093 5.19514 16.9892C5.29746 17.3691 5.5143 17.7082 5.81617 17.9604L12.4336 23.4494C12.5326 23.5316 12.6044 23.6417 12.6397 23.7654C12.675 23.8891 12.6721 24.0205 12.6314 24.1426L9.85121 32.4794C9.72272 32.8649 9.72018 33.2812 9.84396 33.6682C9.96774 34.0552 10.2114 34.3928 10.5397 34.6322C10.8681 34.8715 11.264 35.0003 11.6704 34.9997C12.0767 34.9991 12.4723 34.8693 12.7999 34.6291L19.6241 29.6252C19.7336 29.545 19.8657 29.5018 20.0014 29.5018C20.137 29.5018 20.2691 29.545 20.3786 29.6252L27.2002 34.6278C27.5277 34.8689 27.9235 34.9994 28.3302 35.0005C28.7369 35.0016 29.1334 34.8731 29.4621 34.6338C29.7909 34.3944 30.035 34.0566 30.1589 33.6693C30.2829 33.2819 30.2803 32.8652 30.1515 32.4794L27.3713 24.1375C27.3306 24.0154 27.3277 23.884 27.363 23.7603C27.3983 23.6366 27.4701 23.5265 27.5691 23.4443L34.1993 17.9477C34.4973 17.6933 34.7101 17.3536 34.8089 16.9744C34.9078 16.5953 34.8879 16.1949 34.752 15.8274V15.8274Z" class="primary"></path></svg></span>
                <div class="unnnic-card-marketplace__content__rating__value">4.9</div>
                <div class="unnnic-card-marketplace__content__rating__comments">(undefined some specific text)</div>
              </div>
            </div>
          </div>
        </div>
        <!---->
      </section>
      <!---->
      <div>
        <transition-stub name="fade">
          <!---->
        </transition-stub>
        <!---->
      </div>
    </div>
    <div>
      <section id="app-grid">
        <div class="app-grid__header">
          <div data-v-0231bb98="" class="app-grid__header__icon unnnic-avatar-icon aux-orange-500 sm opacity"><span data-v-4ce83152="" data-v-0231bb98="" class="material-symbols-rounded unnnic-icon-scheme--aux-orange-500 unnnic-icon-size--md material-symbols-rounded--filled"> power </span></div>
          <p class="app-grid__header__title">some specific text</p>
        </div>
        <div class="app-grid__content">
          <div class="unnnic-card-component unnnic-card-marketplace unnnic-card-marketplace__clickable unnnic-card-marketplace__add app-grid__content__item" v-bind="[object Object]" button-action="() => {}" percent="0" info-position="right" enabled="enabled" scheme="aux-blue" has-information-icon="true">
            <div class="unnnic-card-marketplace__icons"><img src="https://weni-sp-push-dev.s3.sa-east-1.amazonaws.com/svg/logo-wwc.svg" class="unnnic-card-marketplace__icons__app">
              <div><button data-v-2d9f2cbf="" disabled="disabled" class="unnnic-button unnnic-button--size-small unnnic-button--secondary unnnic-button--icon-on-center loading-button">
                  <!---->
                  <!----><span data-v-4ce83152="" data-v-2d9f2cbf="" class="unnnic-icon unnnic-icon__size--sm unnnic-icon-scheme--neutral-clean"><svg viewBox="0 0 40 40" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M20 35.0005C19.4825 35.0005 19.0625 34.5805 19.0625 34.063V20.938H5.9375C5.42 20.938 5 20.518 5 20.0005C5 19.483 5.42 19.063 5.9375 19.063H19.0625V5.93799C19.0625 5.42049 19.4825 5.00049 20 5.00049C20.5175 5.00049 20.9375 5.42049 20.9375 5.93799V19.063H34.0625C34.58 19.063 35 19.483 35 20.0005C35 20.518 34.58 20.938 34.0625 20.938H20.9375V34.063C20.9375 34.5805 20.5175 35.0005 20 35.0005Z" fill="#3B414D" class="primary"></path></svg></span><span data-v-2d9f2cbf="" class="unnnic-button__label">  </span>
                  <!----></button>
                <!---->
                <!---->
              </div>
            </div>
            <div class="unnnic-card-marketplace__content">
              <div class="unnnic-card-marketplace__content__title">Weni Web Chat</div>
              <div title="some specific text" class="unnnic-card-marketplace__content__description"> some specific text </div>
              <div class="unnnic-card-marketplace__content__rating"><span data-v-4ce83152="" class="unnnic-card-marketplace__content__rating__star unnnic-icon unnnic-icon__size--ant unnnic-icon-scheme--feedback-yellow"><svg viewBox="0 0 40 40" xmlns="http://www.w3.org/2000/svg"><path d="M34.752 15.8274C34.6157 15.461 34.3708 15.145 34.0501 14.9215C33.7294 14.698 33.3482 14.5776 32.9573 14.5764H25.1872C25.0549 14.5765 24.926 14.5355 24.818 14.4592C24.7101 14.3828 24.6285 14.2748 24.5847 14.1501L21.7955 6.24466C21.6589 5.87927 21.4139 5.56433 21.0934 5.34198C20.7729 5.11964 20.3921 5.00049 20.002 5.00049C19.6119 5.00049 19.2311 5.11964 18.9106 5.34198C18.5901 5.56433 18.3451 5.87927 18.2085 6.24466L18.2021 6.26508L15.4193 14.1501C15.3755 14.2746 15.2942 14.3824 15.1865 14.4588C15.0788 14.5352 14.9501 14.5763 14.8181 14.5764H7.04544C6.65205 14.5761 6.2681 14.697 5.94584 14.9226C5.62357 15.1483 5.37862 15.4677 5.24431 15.8374C5.10999 16.2072 5.09282 16.6093 5.19514 16.9892C5.29746 17.3691 5.5143 17.7082 5.81617 17.9604L12.4336 23.4494C12.5326 23.5316 12.6044 23.6417 12.6397 23.7654C12.675 23.8891 12.6721 24.0205 12.6314 24.1426L9.85121 32.4794C9.72272 32.8649 9.72018 33.2812 9.84396 33.6682C9.96774 34.0552 10.2114 34.3928 10.5397 34.6322C10.8681 34.8715 11.264 35.0003 11.6704 34.9997C12.0767 34.9991 12.4723 34.8693 12.7999 34.6291L19.6241 29.6252C19.7336 29.545 19.8657 29.5018 20.0014 29.5018C20.137 29.5018 20.2691 29.545 20.3786 29.6252L27.2002 34.6278C27.5277 34.8689 27.9235 34.9994 28.3302 35.0005C28.7369 35.0016 29.1334 34.8731 29.4621 34.6338C29.7909 34.3944 30.035 34.0566 30.1589 33.6693C30.2829 33.2819 30.2803 32.8652 30.1515 32.4794L27.3713 24.1375C27.3306 24.0154 27.3277 23.884 27.363 23.7603C27.3983 23.6366 27.4701 23.5265 27.5691 23.4443L34.1993 17.9477C34.4973 17.6933 34.7101 17.3536 34.8089 16.9744C34.9078 16.5953 34.8879 16.1949 34.752 15.8274V15.8274Z" class="primary"></path></svg></span>
                <div class="unnnic-card-marketplace__content__rating__value">4.9</div>
                <div class="unnnic-card-marketplace__content__rating__comments">(undefined some specific text)</div>
              </div>
            </div>
          </div>
        </div>
        <!---->
      </section>
      <!---->
      <div>
        <transition-stub name="fade">
          <!---->
        </transition-stub>
        <!---->
      </div>
    </div>
    <div>
      <section id="app-grid">
        <div class="app-grid__header">
          <div data-v-0231bb98="" class="app-grid__header__icon unnnic-avatar-icon aux-red-500 sm opacity"><span data-v-4ce83152="" data-v-0231bb98="" class="material-symbols-rounded unnnic-icon-scheme--aux-red-500 unnnic-icon-size--md material-symbols-rounded--filled"> bar_chart_4_bars </span></div>
          <p class="app-grid__header__title">some specific text</p>
        </div>
        <div class="app-grid__content">
          <div class="unnnic-card-component unnnic-card-marketplace unnnic-card-marketplace__clickable unnnic-card-marketplace__edit app-grid__content__item" v-bind="[object Object]" button-action="() => {}" percent="0" info-position="right" enabled="enabled" scheme="aux-blue" has-information-icon="true">
            <div class="unnnic-card-marketplace__icons"><img src="[object Object]" class="unnnic-card-marketplace__icons__app"></div>
            <div class="unnnic-card-marketplace__content">
              <div class="unnnic-card-marketplace__content__title">Power BI</div>
              <div title="some specific text" class="unnnic-card-marketplace__content__description"> some specific text </div>
              <!---->
            </div>
          </div>
        </div>
        <!---->
      </section>
      <!---->
      <div>
        <transition-stub name="fade">
          <!---->
        </transition-stub>
        <!---->
      </div>
    </div>
  </div>
  <!---->
  <!---->
</div>
`;
